packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -262,40 +265,135 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
+  _RenderEditableCustomPaint? _backgroundRenderObject;
+
+  RenderEditablePainter? _foregroundPainter;
+  void _updateForgroundPainter(RenderEditablePainter? newPainter) {
+    final _CompositeRenderEditablePainter effectivePainter = newPainter == null
+      ? _builtInForegounrdPainters
+      : _CompositeRenderEditablePainter(painters: <RenderEditablePainter>[
+         _builtInForegounrdPainters,
+         newPainter,
+      ]);
+
+    if (_foregroundRenderObject == null) {
+      final _RenderEditableCustomPaint foregroundRenderObject = _RenderEditableCustomPaint(painter: effectivePainter);
+      adoptChild(foregroundRenderObject);
+      _foregroundRenderObject = foregroundRenderObject;
+    } else {
+      _foregroundRenderObject?.painter = effectivePainter;
+    }
+    _foregroundPainter = newPainter;
   }
+  /// Sets the [RenderEditablePainter] to use for painting above this
+  /// [RenderEditable]'s text content.
+  ///
+  /// The new [RenderEditablePainter] will replace the previously specified
+  /// foreground painter, and schedule a repaint if the new painter's
+  /// `shouldRepaint` method returns true.
+  void setForegroundPainter(RenderEditablePainter? newPainter) {
+    if (newPainter == _foregroundPainter)
+      return;
+    _updateForgroundPainter(newPainter);
+  }
+
+  RenderEditablePainter? _painter;
+  void _updatePainter(RenderEditablePainter? newPainter) {
+    final _CompositeRenderEditablePainter effectivePainter = newPainter == null
+      ? _builtInPainters
+      : _CompositeRenderEditablePainter(painters: <RenderEditablePainter>[_builtInPainters, newPainter]);
 
+    if (_backgroundRenderObject == null) {
+      final _RenderEditableCustomPaint backgroundRenderObject = _RenderEditableCustomPaint(painter: effectivePainter);
+      adoptChild(backgroundRenderObject);
+      _backgroundRenderObject = backgroundRenderObject;
+    } else {
+      _backgroundRenderObject?.painter = effectivePainter;
+    }
+    _painter = newPainter;
+  }
+  /// Sets the [RenderEditablePainter] to use for painting beneath this
+  /// [RenderEditable]'s text content.
+  ///
+  /// The new [RenderEditablePainter] will replace the previously specified
+  /// painter, and schedule a repaint if the new painter's `shouldRepaint`
+  /// method returns true.
+  void setPainter(RenderEditablePainter? newPainter) {
+    if (newPainter == _painter)
+      return;
+    _updatePainter(newPainter);
+  }
+
+  // Caret Painters:
+  // The floating painter. This painter paints the regular caret as well.
+  late final _FloatingCursorPainter _caretPainter = _FloatingCursorPainter(_onCaretChanged);
+
+  // Text Highlight painters:
+  final _TextHighlightPainter _selectionPainter = _TextHighlightPainter();
+  final _TextHighlightPainter _autocorrectHighlightPainter = _TextHighlightPainter();
+
+  _CompositeRenderEditablePainter get _builtInForegounrdPainters => _cachedBuiltInForegroundPainters ??= _createBuiltInForegroundPainters();
```

Comment:
Also, you should be able to just defines this as:

```dart
late final _CompositeRenderEditablePainter _builtInForegounrdPainters = _createBuiltInForegroundPainters();
```

With late it will only get instantiated on first access.

Reply:
Oh, never mind. I see below you need the option to clear this.